lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1653,6 +1669,255 @@ def mock_certificate_downloadable_status(
         }
 
 
+# pylint: disable=protected-access, no-member
+@ddt.ddt
+class ProgressPageShowCorrectnessTests(ProgressPageBaseTests):
+    """
+    Tests that verify that the progress page works correctly when displaying subsections where correctness is hidden.
+    """
+    NOW = datetime.now(UTC)
+    DAY_DELTA = timedelta(days=1)
+    YESTERDAY = NOW - DAY_DELTA
+    TOMORROW = NOW + DAY_DELTA
+
+    def setUp(self):
+        super(ProgressPageShowCorrectnessTests, self).setUp()
+        self.staff_user = UserFactory.create(is_staff=True)
+
+    def setup_course(self, show_correctness='', due_date=None, graded=False, add_problem=True, **course_options):
+        """
+        Set up course with a subsection with the given show_correctness, due_date, and graded settings.
```

Comment:
Went one better, and separated it out into its own method:  https://github.com/edx/edx-platform/pull/15018/commits/d7e44c50acc6dc3a73fc400f410cfe4ceaca834d#diff-8aefe502dfa8ee5a3e18c9227e61b034R1732

When I started writing these tests, I wasn't very familiar with how the modulestore test objects worked, and so didn't know how to update an existing course.  But lo, there's a way!